charts/tempo-distributed/templates/_helpers.tpl

```diff
@@ -120,3 +161,132 @@ Return if ingress supports pathType.
 {{- define "tempo.ingress.supportsPathType" -}}
   {{- or (eq (include "tempo.ingress.isStable" .) "true") (and (eq (include "tempo.ingress.apiVersion" .) "networking.k8s.io/v1beta1") (semverCompare ">= 1.18-0" .Capabilities.KubeVersion.Version)) -}}
 {{- end -}}
+
+{{/*
+Resource name template
+*/}}
+{{- define "tempo.resourceName" -}}
+{{ include "tempo.fullname" .ctx }}{{- if .component -}}-{{ .component }}{{- end -}}
+{{- end -}}
+
+{{/*
+Calculate the config from structured and unstructred text input
+*/}}
+{{- define "tempo.calculatedConfig" -}}
+{{ tpl (mergeOverwrite (tpl .Values.config . | fromYaml) .Values.tempo.structuredConfig | toYaml) . }}
+{{- end -}}
+
+{{/*
+Calculate the overrides config from structured and unstructred text input
+*/}}
+{{- define "tempo.calculatedOverridesConfig" -}}
+{{ tpl .Values.overrides . }}
+{{- end -}}
+
+{{/*
+The volume to mount for tempo configuration
+*/}}
+{{- define "tempo.configVolume" -}}
+{{- if eq .Values.configStorageType "Secret" -}}
+secret:
+  secretName: {{ tpl .Values.externalConfigSecretName . }}
+{{- else if eq .Values.configStorageType "ConfigMap" -}}
+configMap:
+  name: {{ tpl .Values.externalConfigSecretName . }}
+  items:
+    - key: "tempo.yaml"
+      path: "tempo.yaml"
+    - key: "overrides.yaml"
+      path: "overrides.yaml"
+{{- end -}}
+{{- end -}}
+
+{{/*
+Internal servers http listen port - derived from Loki default
+*/}}
+{{- define "tempo.serverHttpListenPort" -}}
+{{ (((.Values.tempo).structuredConfig).server).http_listen_port | default "3100" }}
+{{- end -}}
+
+{{/*
+Internal servers grpc listen port - derived from Tempo default
+*/}}
+{{- define "tempo.serverGrpcListenPort" -}}
+{{ (((.Values.tempo).structuredConfig).server).grpc_listen_port | default "9095" }}
+{{- end -}}
+
+{{/*
+Memberlist bind port
+*/}}
+{{- define "tempo.memberlistBindPort" -}}
+{{ (((.Values.tempo).structuredConfig).memberlist).bind_port | default "7946" }}
+{{- end -}}
+
+{{/*
+Calculate values.yaml section name from component name
```

Comment:
fyi, it turned out that we didn't have consistent naming , so we ended up changing this to have a map in it and return the section, instead of its name: https://github.com/grafana/mimir/blob/8962290b64ebcc4c9ba91a8ff1455f7a600c3a29/operations/helm/charts/mimir-distributed/templates/_helpers.tpl#L378